aiohttp/helpers.py

```diff
@@ -797,9 +797,40 @@ def set_result(fut: "asyncio.Future[_T]", result: _T) -> None:
         fut.set_result(result)
 
 
-def set_exception(fut: "asyncio.Future[_T]", exc: BaseException) -> None:
-    if not fut.done():
-        fut.set_exception(exc)
+_EXC_SENTINEL = BaseException()
+
+
+class ErrorableProtocol(Protocol):
+
+    def set_exception(
+        self,
+        exc: BaseException,
+        exc_cause: BaseException = _EXC_SENTINEL,
+    ) -> None:
+        raise NotImplementedError
```

Comment:
```suggestion
        exc_cause: BaseException = ...,
    ) -> None:
        ...
```

You probably missed my comment as you posted a second after, but this is a stub.
https://github.com/aio-libs/aiohttp/pull/8089#discussion_r1469763839

Reply:
Ah, I wasn't sure that was mandatory.

Reply:
Well, no, but it makes it clearer at a glance and avoids maintenance of things which are literally ignored and never run. The _EXC_SENTINEL is certainly incorrect, as no implementation of this method is defaulting to that parameter (but, as I say it's ignored by mypy as it's nothing to do with the type).

Reply:
> The _EXC_SENTINEL is certainly incorrect, as no implementation of this method is defaulting to that value (but, as I say it's ignored by mypy as it's nothing to do with the type).

That's not the case — I explicitly import that sentinel and use it as the default in all the signatures. This is the only use for that import outside this module.

Reply:
Ah, yes, I didn't recheck the diff, thought the methods were left unchanged.

Still, it'd look a little odd to me for a stub. Up to you.

Reply:
@Dreamsorcerer if we have the sentinel default value there in the stub, would MyPy enforce that the method implementations also have the same default value?

Reply:
No, I'm pretty sure mypy completely ignores the values, as it's not typing information. The only benefit might be in IDEs showing the default value, which is why typeshed have started including simple defaults. But, I don't think that's relevant to a Protocol.

Reply:
It'd be nice to enforce this, I suppose. But alright, I'll use the ellipsis.